amr-wind/overset/TiogaInterface.cpp

```diff
@@ -235,7 +235,7 @@ void TiogaInterface::amr_to_tioga_mesh()
         ngrids_global += mesh.boxArray(lev).size();
 
         const auto& dmap = mesh.DistributionMap(lev);
-        for (long d = 0; d < dmap.size(); ++d) {
+        for (size_t d = 0; d < static_cast<size_t>(dmap.size()); ++d) {
```

Comment:
maybe we don't need the static_cast<size_t> ... I think .size() returns that anyways?

Reply:
It actually returns a `long` from what the compiler is telling me. So if `d` stayed as a `long`, it would propagate narrowing conversions to the code within the loop.

Reply:
So I felt promoting the indexing to `size_t` from `long` was the safest choice?

Reply:
Without these changes:
```
amr-wind/overset/TiogaInterface.cpp:239:22: warning: narrowing conversion from 'long' to signed type 'int' is implementation-defined [bugprone-narrowing-conversions]
            if (dmap[d] == iproc) {
                     ^
```

Reply:
this won't work? 
`for (size_t d = 0; d < dmap.size(); ++d) {`

Reply:
No
```
amr-wind/overset/TiogaInterface.cpp:238:30: warning: comparison of integers of different signs: 'size_t' (aka 'unsigned long') and 'amrex::Long' (aka 'long') [clang-diagnostic-sign-compare]
        for (size_t d = 0; d < dmap.size(); ++d) {
                             ^
```

Reply:
So if d is `long`, it has to be cast to an `int` in the array index, otherwise, promoting everything to `size_t` works. It's strange to me that `long` appears to not work as an array index, yet `int` and `size_t` (or `unsigned long`) seem to not cause warnings.

Reply:
So right now the code implicitly casts `d` to an `int`. I could just explicitly do that instead of these changes.

Reply:
I'm going to remove these changes from this PR to simplify it.